src/graphics/device.js

```diff
@@ -466,7 +466,9 @@ Object.assign(pc, function () {
             scopeY = value[1];
             scopeZ = value[2];
             if (uniformValue[0] !== scopeX || uniformValue[1] !== scopeY || uniformValue[2] !== scopeZ) {
-                gl.uniform3fv(uniform.locationId, value);
+                // A shader can declare color as float3, thus direct setting of all components will fail - 
+                // let's keep it safe and set only 3 float array
+                gl.uniform3fv(uniform.locationId, [ scopeX, scopeY, scopeZ ]);
```

Comment:
I might say something meaningless, but shouldn't the same be done for other functions like this one in the file?

Reply:
Technically, yes - but I have avoided that intentionally. It doesn't seem right to pass vec4 to a vec3 uniform (because, why, right?), but this is heavily used in Unity, so we kinda have to have this.